server/api/service_gc_safepoint.go

```diff
@@ -63,11 +65,31 @@ func (h *serviceGCSafepointHandler) GetGCSafePoint(w http.ResponseWriter, _ *htt
 	}
 
 	ssps := make([]*endpoint.ServiceSafePoint, 0, len(gcState.GCBarriers))
+	var gcWorkerSsp *endpoint.ServiceSafePoint
 	for _, barrier := range gcState.GCBarriers {
-		ssps = append(ssps, barrier.ToServiceSafePoint(constant.NullKeyspaceID))
+		equivalentSsp := barrier.ToServiceSafePoint(constant.NullKeyspaceID)
+		ssps = append(ssps, equivalentSsp)
+		if equivalentSsp.ServiceID == keypath.GCWorkerServiceSafePointID {
+			gcWorkerSsp = equivalentSsp
+		}
+	}
+	if gcWorkerSsp == nil {
+		// Generate a pseudo service safe point for GC worker. While GC worker's service safe point won't exist after
+		// the new GC states API is adopted, some existing tests relies on the fact to pass: the minimal service safe
+		// point indicates the lower bound of the timestamp that is safe to read. Now, this value becomes the txn safe
+		// point. We generate a pseudo service safe point with the value of txn safe point to provide the compatibility.
```

Comment:
Will it be removed later if we introduce the new pd-ctl cmd?

Reply:
If necessary, we can remove the service-gc-safepoint API entirely. But I think it doesn't make sense to remove the compatibility logic while keeping the command.